### PR TITLE
fix to work with python3

### PR DESCRIPTION
**Pull request by meduz**
tries to solve https://github.com/Zulko/moviepy/issues/160

tested with python  2.7.9 and 3.4.3

(sorry my editor automatically "cleaned" spaces - I fixed that)

### CODE REVIEW

**Comment by Zulko** [comment]
Hey there, can you explain the changes in mplfig_to_npimage ?

**Comment by meduz** [comment]
To port moviepy's code to python 3, a difference is that strings are now represented as unicode and data as bytes while this was different (and sometimes problematic) in python 2:

https://docs.python.org/3.3/whatsnew/3.0.html#text-vs-data-instead-of-unicode-vs-8-bit

This was already implemented in matplotlib, trying to make this backportable with python 2.7, see for instance

https://github.com/matplotlib/matplotlib/commit/f4fee350f9fbc639853bee76472d8089a10b40bd

in fact, there is no change to the API, but the correct backend has to be called.

basically, what I am doing here is to call for the correct backend (Agg) before calling the same functions and return a numpy array.

hope it is a bit clearer, I am still learning my way to help contribute to the libraries that I use.

**Comment by meduz** [comment]
I still have problems rendering movies in a notebook using python 3 - see http://blog.invibe.net/posts/2015-04-26-moviepy-and-python3.html - do you have the same problem?
cheers!

**Comment by Zulko** [comment]
I didn't have time to look into this (im still on Python 2 in my notebooks), now its unclear, does the commit actually solves the problem ?

**Comment by meduz** [comment]
Yes the commit solves the problem in py3k - and should be back-portable with python 2.7 and higher.

**Comment by flothesof** [comment]
Hi everyone,

I would like to add that I have encountered the same problem than @meduz. Would it be possible to accept the PR so as to fix this problem with Python 3?

Best regards,

Florian

**Comment by Krastanov** [comment]
Any progress on this?

**Comment by Zulko** [comment]
Sorry for being slow. Too bad moviepy doesn't have a test suite.
